lib/resources/git_resource.py

```diff
@@ -27,90 +30,159 @@ def __init__(self, database: DbConnector):
+
+        # Iterating over commits & tags
+        logger.info("Retrieving diff of all tags ...")
+        for idx, (commit, tag) in enumerate(tag_commit_list):
+            commit_diff: DiffIndex = None
+
+            if idx + 1 < len(tag_commit_list):
+                commit_diff = commit.diff(tag_commit_list[idx + 1][0])
+                logger.debug(f"Comparing {tag.name} with {tag_commit_list[idx + 1][1].name}")
+            else:
+                commit_diff = commit.diff(tag_commit_list[idx - 1][0])
+                logger.debug(f"Comparing {tag.name} with {tag_commit_list[idx - 1][1].name}")
+
+            for diffidx in commit_diff:
+                diffidx: Diff = diffidx
+
+                if diffidx.a_blob:
+                    logger.debug((
+                        f"Change type: {diffidx.change_type}    File path: {diffidx.a_blob.path}    Tag: {tag.name}"
+                        f"    Blob a: {diffidx.a_blob}    Commit hash: {commit.hexsha}"
+                    ))
+                    files.append((diffidx.a_blob.path, tag.name, commit.hexsha, diffidx.a_blob.hexsha))
+                elif diffidx.b_blob:
+                    logger.debug((
+                        f"Change type: {diffidx.change_type}    File path: {diffidx.b_blob.path}    Tag: {tag.name}"
+                        f"    Blob b: {diffidx.b_blob}    Commit hash: {commit.hexsha}"
+                    ))
+                    files.append((diffidx.b_blob.path, tag.name, commit.hexsha, diffidx.b_blob.hexsha))
+        return files
+
+    @classmethod
+    def _get_tag_files(cls, tag_commit: Tuple[Commit, Tag]) -> List[Tuple[str, str, str, str]]:
+        """
+        This method retrieves all files in a tag.
+        """
+        files = []
 
-                    self._database.insert_or_update_hash(session, hash_value, technology, tag)
+        for (file_path, file_hex) in GitResource.get_all_files_from_commit(tag_commit[0]):
+            logger.debug((
+                f"File path: {file_path}    Tag: {tag_commit[1].name}"
+                f"    Commit hash: {tag_commit[0].hexsha}"
+            ))
+            files.append((file_path, tag_commit[1].name, tag_commit[0].hexsha, file_hex))
+        return files
 
-    def clone_checkout_and_compute_hashs(self, session_scope, url):
+    def clone_checkout_and_compute_hashs(self, session_scope, url: str):
         """
-        This method clones the repository from url, retrieves and stores tags in the database,
-        then checkout each tag, sotres files information in the database and computes the hashs.
+        This method clones the repository from url, retrieves tags, compares each tags to retrieve only modified files,
+        computes their hashes and then stores the tags & files information in the database.
         """
-        technology = url.split('.git')[0].split('/')[-1]
         with tempfile.TemporaryDirectory() as tmp_dir_name:
+            technology = url.split('.git')[0].split('/')[-1]
```

Comment:
`technology` is not needed in this `with` block. Maybe you can move this before the next `with` block  ?